lib/active_model/serializer.rb

```diff
@@ -182,7 +182,7 @@ def as_json(adapter_opts = nil)
 
     # Used by adapter as resource root.
     def json_key
```

Comment:
From what I see, we can set the `root` as adapter option and call `serializer._type` and `serializer.object.class.model_name.to_s.underscore` from the adapter. Same goes for collection_serializer. What do you think?